cve_bin_tool/cvedb.py

```diff
@@ -286,13 +286,13 @@ def init_database(self):
             versionStartIncluding TEXT,
             versionStartExcluding TEXT,
             versionEndIncluding TEXT,
-            versionEndExcluding TEXT
+            versionEndExcluding TEXT,
+            PRIMARY KEY (cve_number, vendor, product, version),
```

Comment:
This still won't work for the `cve_range` table -- many entries have multiple ranges.  Honestly, I thnk we have three options here:

1. cve primary key is every single row in the table for cve_range
2. don't set a primary key for cve_range
3. add a primary_key column and auto-generate a key

I went with #2 intentionally because we didn't have any use cases where we needed a primary key for that table: we're always searching by cve_number. I'm willing to be convinced otherwise if we get a performance boost for having a key, but I couldn't find much indication that this would be true.  (I know, i know, it's "best practices" for real databases, but sqlite is basically a flat file with pretentions, so it didn't seem worth generating data that wasn't used anywhere.)

We still want the index on this, and the foreign key stuff is a good idea, though.

Reply:
Yes, you are right there are some rows where my assigned primary won't be able to identify a row uniquely. I have run this query and found out there are approx 10000 CVEs with this issue:
```sql
select cve_number, vendor, product, version, count(*) as cnt from cve_range group by cve_number, vendor, product, version having cnt > 1;
```
This problem seems to be originating due to denormalized data. In this case table without the primary key seems the right choice. Although some indexing can help to organize the database efficiently.